### PR TITLE
Fix handling nullptr when cleaning up during ReJIT

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**
It turns out on GetReJITParameters() method end, ~CRemoveMethodInfoAfterRejit invokes CModuleInfo::SetRejitMethodInfo with pMethodInfo as nullptr. The SetRejitMethodInfo does not remove the MethodToken from the map when setting the nullptr.

This addresses #197

### CODE REVIEW

File: src/InstrumentationEngine/ModuleInfo.cpp [hunk 1]
```diff
@@ -907,14 +907,22 @@ HRESULT MicrosoftInstrumentationEngine::CModuleInfo::SetRejitMethodInfo(_In_ mdT
...
+            CLogging::LogMessage(_T("CModuleInfo::SetRejitMethodInfo - adding new method token %04x."), methodToken);
+        }
 
-    m_methodInfosByToken[methodToken] = pMethodInfo;
+        m_methodInfosByToken[methodToken] = pMethodInfo;
+    }
+    else
+    {
+        CLogging::LogMessage(_T("CModuleInfo::SetRejitMethodInfo - erasing existing method token %04x."), methodToken);
+        m_methodInfosByToken.erase(methodToken);
```
Comment (WilliamXieMSFT) [inline src/InstrumentationEngine/ModuleInfo.cpp]: >erase [](start = 29, length = 5)

I'll add another check to ensure the MethodToken exists before erasing

Comment (WilliamXieMSFT) [reply src/InstrumentationEngine/ModuleInfo.cpp]: erase will return 0 if the element does not exist

---
In reply to: [375417655](https://github.com/microsoft/CLRInstrumentationEngine/pull/200#discussion_r375417655) [](ancestors = 375417655)

**Comment by sanikolov** [comment]
confirmed fixed, thanks

**Pull request merged**